File: 02 medium/array/49_group_anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    let result = []
    let res_map = {}
    for (var i = 0; i < strs.length; i++) {
        let word = strs[i].split('').sort().join('');
        if (res_map[word]) res_map[word].push(strs[i])
        else res_map[word] = [strs[i]]
    }
    for (var m in res_map) {
        result.push(res_map[m])
    }

    return result
};

File: array/49_group_anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        result = []
        a_map = defaultdict(list)

        for s in strs:
            a_map["".join(sorted(s))].append(s)
        
        for k in a_map:
            result.append(a_map[k])
        
        return result
